src/main.py

```diff
@@ -1,8 +1,64 @@
-from fastapi import FastAPI
+import uuid
+
+from fastapi import FastAPI, HTTPException
+from src.schemas import ProjectDetails, Project
 
 app = FastAPI()
+projects: list[ProjectDetails] = []
+
+
+def get_project(project_id: uuid.UUID) -> ProjectDetails:
+    project = next((p for p in projects if p.project_id == project_id), None)
+    if project is None:
+        raise HTTPException(status_code=404, detail=f"ProjectDetails {project_id} not found")
+    return project
+
+
+def create_project_(project: Project) -> ProjectDetails:
+    new_project = ProjectDetails(**project.model_dump())
+    projects.append(new_project)
+    return new_project
+
+
+def get_projects_() -> list[ProjectDetails]:
+    return projects
+
+
+def update_project_details_(project_id: uuid.UUID, name: str, description: str | None) -> ProjectDetails:
+    project = get_project(project_id)
```

Comment:
same here - function must take project instance, not id. You have implemented retrieval in another function